packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -194,14 +194,41 @@ function safelyDetachRef(current: Fiber, nearestMountedAncestor: Fiber) {
   if (ref !== null) {
     if (typeof ref === 'function') {
       if (__DEV__) {
-        invokeGuardedCallback(null, ref, null, null);
+        if (
+          enableProfilerTimer &&
+          enableProfilerCommitHooks &&
+          current.mode & ProfileMode
+        ) {
+          try {
+            startLayoutEffectTimer();
+            invokeGuardedCallback(null, ref, null, null);
```

Comment:
`invokeGuardedCallback` won't throw so you  don't need the try/finally

Reply:
Unless `startLayoutEffectTimer` throws?

Reply:
lol duh